### PR TITLE
Redux in react components

### PR DESCRIPTION
**Pull request by hopeforkoami**
In this milestone I have
 -  Understood the concept of state management.
 -  Used store, actions, and reducer in React
 -  Connected Redux and React app
 -  Used Redux immutability
 -  Imported Redux Provider and store in the main component React app.
 -  Used useSelector to consume the state.
 -  Imported Redux useDispatch for removing and adding functionalities to dispatch respective actions ADD_BOOK and REMOVE_BOOK

### CODE REVIEW

File: src/Components/BookList.js [hunk 1]
```diff
@@ -3,14 +3,15 @@ import PropTypes from 'prop-types';
...
+  const { books, removeHandler } = props;
   return (
     <ul className="booklist">
       {
           books.map((bookElment) => (
             <Book
               key={bookElment.id}
               bookdata={bookElment}
+              removeBookFromList={removeHandler}
             />
```
Comment (GzimAsani) [inline src/Components/BookList.js:15]: - [ ] List of books need to  display books from Redux store by using useSelector function.

Comment (GzimAsani) [reply src/Components/BookList.js:15]: - [ ] On category page please use redux that when the button is clicked it displays **under construction**

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello  @hopeforkoami                          :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with the project, I really liked how you divided your work on components :+1: 

 ## Required Changes 
- [ ] Check the comments below :arrow_double_down: 




 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**